owncloudApp/src/main/java/com/owncloud/android/ui/activity/ShareActivity.kt

```diff
@@ -0,0 +1,394 @@
+        if (prev != null) {
+            ft.remove(prev)
+        }
+        ft.addToBackStack(null)
+
+        // Create and show the dialog
+        val newFragment = PublicShareDialogFragment.newInstanceToCreate(
+            file,
+            account,
+            defaultLinkName
+        )
+        newFragment.show(ft, TAG_PUBLIC_SHARE_DIALOG_FRAGMENT)
+    }
+
+    override fun showEditPublicShare(share: OCShare) {
+        val ft = supportFragmentManager.beginTransaction()
+        val prev = supportFragmentManager.findFragmentByTag(TAG_PUBLIC_SHARE_DIALOG_FRAGMENT)
+        if (prev != null) {
+            ft.remove(prev)
+        }
+        ft.addToBackStack(null)
+
+        // Create and show the dialog.
+        val newFragment = PublicShareDialogFragment.newInstanceToUpdate(
+            file, share,
+            account
+        )
+        newFragment.show(ft, TAG_PUBLIC_SHARE_DIALOG_FRAGMENT)
+    }
+
+    override fun copyOrSendPublicLink(share: OCShare) {
+        fileOperationsHelper.copyOrSendPublicLink(share)
+    }
+
+    /**
+     * Updates the view associated to the activity after the finish of some operation over files
+     * in the current account.
+     *
+     * @param operation Removal operation performed.
+     * @param result    Result of the removal.
+     */
+    override fun onRemoteOperationFinish(operation: RemoteOperation<*>, result: RemoteOperationResult<*>) {
+        super.onRemoteOperationFinish(operation, result)
+
+        if (result.isSuccess || operation is GetSharesForFileOperation && result.code == RemoteOperationResult.ResultCode.SHARE_NOT_FOUND) {
+            Log_OC.d(TAG, "Refreshing view on successful operation or finished refresh")
+            refreshSharesFromStorageManager()
+            if (operation is GetSharesForFileOperation) {
+                mGetSharesForFileAsyncTask = null
+            }
+        }
+
+        if (operation is CreateShareViaLinkOperation) {
+            onCreateShareViaLinkOperationFinish(operation, result as RemoteOperationResult<ShareParserResult>)
+        }
+
+        if (operation is UpdateShareViaLinkOperation) {
+            onUpdateShareViaLinkOperationFinish(operation, result as RemoteOperationResult<ShareParserResult>)
+        }
+
+        if (operation is RemoveShareOperation && result.isSuccess && editShareFragment != null) {
+            supportFragmentManager.popBackStack()
+        }
+
+        if (operation is UpdateSharePermissionsOperation
+            && editShareFragment != null && editShareFragment!!.isAdded
+        ) {
+            editShareFragment!!.onUpdateSharePermissionsFinished(result as RemoteOperationResult<ShareParserResult>?)
+        }
+    }
+
+    private fun onCreateShareViaLinkOperationFinish(
+        operation: CreateShareViaLinkOperation,
+        result: RemoteOperationResult<ShareParserResult>
+    ) {
+        if (result.isSuccess) {
```

Comment:
I'd rather be using 

if (!result.isSuccess) {
  // show error
 return;
}

// success block

It can be applied for all cases in this file

Reply:
👍